ListLibrary/MyList.cs

```diff
@@ -0,0 +1,590 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace ListLibrary
+{
+    public class MyList<T> : IMyList<T> where T : IComparable<T>
+    {
+        private const int _defaultCapacity = 4;
+        private const double _capacityRise = 1.33;
+
+        private T[] _items;
+        private int _size;
+
+        static readonly T[] _emptyArray = new T[0];
+
+        #region Constructors, Propeerties and Interfaces implementation
+
+        public MyList()
+        {
+            _items = _emptyArray;
+        }
+
+        public MyList(int capacity)
+        {
+            if (capacity < 0)
+            {
+                throw new ArgumentException("Capacity should be >= 0");
+            }
+
+            if (capacity == 0)
+            {
+                _items = _emptyArray;
+
+            }
+            else
+            {
+                _items = new T[capacity];
+            }
+        }
+
+        public MyList(IEnumerable<T> collection)
```

Comment:
Try to send null here. And empty collection too.

Reply:
Empty collection is ok. Foreach will add nothing